### PR TITLE
expression: rewrite builtin string function: INSTR

### PR DESCRIPTION
**Pull request by zz-jason**
to https://github.com/pingcap/tidb/issues/4080

### CODE REVIEW

**Comment by zz-jason** [comment]
@breeswish @XuHuaiyu PTAL

File: expression/builtin_string.go [hunk 1]
```diff
@@ -2396,58 +2397,69 @@ type instrFunctionClass struct {
...
-	sig := &builtinInstrSig{newBaseBuiltinFunc(args, ctx)}
-	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpInt, tpString, tpString)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf.tp.Flen = 11
```
Comment (XuHuaiyu) [inline expression/builtin_string.go]: why set 11 here?
ported from mysql?

**Comment by XuHuaiyu** [comment]
Rest LGTM

File: expression/builtin_string.go (hunk 1 above)
Comment (zz-jason) [inline expression/builtin_string.go]: yes

**Pull request merged**